aten/src/ATen/native/BinaryOps.cpp

```diff
@@ -121,6 +122,23 @@ Tensor& add_relu_(Tensor& self, const Tensor& other, Scalar alpha) {
   return add_relu_impl(self, self, other, alpha);
 }
 
+Tensor& copysign_out(Tensor& result, const Tensor& self, const Tensor& other) {
+  auto iter = TensorIterator::binary_float_op(result, self, other);
```

Comment:
I don't think this is a binary float op. For example, copysign(int32, int32)->int32, not ->float32.

Reply:
In Numpy, actually copysign(int32, int32)->float64. And, there is no integral copysign both in Numpy and POSIX.

Reply:
That's true but I don't think we should support NumPy's behavior at this time. The cpp standard library doesn't implement copysign for any types other than float or double, but it seems easy to extend the definition to integer types and other float types (extending it to complex types might be unintuitive). It's weird to me that we would promote integer types, however, since the output of the function is representable in the input dtype. Typically int->float type promotion only occurs when the output cannot be represented in the inputs dtype. For example, sin(int)->float.

Reply:
It's not just NumPy's behavior though; Python copysign also promotes ints to floats:
```
>>> import math
>>> math.copysign(1, 1)
1.0
```
Folks who use torch.copysign may expect it to work in the same way as these other libraries; to avoid confusion we could either have it be consistent or disable it for integral types

Reply:
My proposal was to disable it for integer types.